### DISCUSSION TITLE
Frequência ideal para mudanças de emprego

### DISCUSSION THREAD

**Discussion by JonatasRicardo** [frontendbr/forum#1114] [Discussão]
Pessoal, qual é a frequência ideal para mudanças de emprego?

Vejo alguns casos onde os desenvolvedores passam de 3 a 9 meses em cada empresa.
Outros que chegam a ficar mais de 4 anos em um mesmo lugar.

- O que é melhor para a carreira?
- Quais os motivos que levam vocês a ficar pouco ou muito tempo em um mesmo lugar?
- Em suas trajetórias, quais foram as consequências das escolhas que foram feitas?

No meu caso eu já cheguei a ficar 4 anos e meio em um lugar e 7 meses em outro. No período mais longo tive um ganho incrível em soft skills, no mais curto tive um ótimo salto de remuneração.

**Comment by mahenrique94** [comment]
Acho que quantos menos você mudar, mais segurança e qualidade seu currículo terá.

Pessoas que não param em um único serviço, para mim como uma empresa, não seria interessante.

Acho que você tem que estar feliz e bem onde trabalha, salário também deve ser coerente com sua função (não deve ser o único fator). Seu trabalho também deve oferecer uma carreira, crescimento e evolução também, não adianta ficar anos fazendo e ganhando o mesmo.

Claro que novas oportunidades e desafios, sempre são bem vindos.

**Comment by kivervinicius** [comment]
rapaz, eu não sei se tem algo ideal, quando vc acha a empresa que te da um bom feedback que se sente "em casa" é dificil querer sair, porém tem o fator zona de conforto. Bom já faz 7 anos que estou na mesma empresa e fiquei mais 6 anos na anterior haha

**Comment by kelciocajueiro** [comment]
Tem o fator outsourcing também. Eu já mudei muito de empresa por causa de término de contrato com o cliente.

**Comment by kvnol** [comment]
Não que seja certo, mas algo que levo pra mim (que não gosto de rotina) é 2 anos, no máximo.

**Comment by lucasfcosta** [comment]
⚠️ **ALERTA DE MINHA OPINIÃO**™️ 

There is no such thing as "tempo ideal".

O tempo ideal é o tempo em que você ainda esteja gostando do que faz e sendo capaz de desenvolver suas habilidades como engenheiro.

Se uma empresa olhar pro seu currículo e te julgar porque você ficou pouco tempo em algumas empresas ou tempo demais em outras é porque ela não é uma boa empresa. É claro que existem exceções, como por exemplo se você teve 4 empregos e ficou 2 meses em todos.

Eu já trabalhei 2 anos em alguns lugares e 3 meses em outros e isso nunca me causou problemas.

Você precisa simplesmente ser um bom profissional e entregar suas tasks com qualidade, nada mais.

Só existe uma coisa que interessa quando se trata de um emprego: o quanto você custa vs. o quanto de valor você gera. Bem-vindo ao capitalismo.

Quanto mais valor você gera, mais caro é o seu tempo.

Eu sempre digo que as pessoas devem focar mais em se tornar melhores profissionais e menos em fazer currículo. O que importa no final são os seus accomplishments e o quanto de valor você gerou nos locais onde passou.

E assim como as empresas selecionam seus empregados, você deve selecionar as empresas. Toda relação é uma via de mão dupla. Se a empresa espera que você seja flexível com seu tempo de trabalho ela também vai ter que entender quando você precisar que ela seja flexível com as horas que requer você lá dentro, assim como quanto mais ela te paga, mais valor ela espera que você gere.

Enfim, sendo bem realista é isso, vou parar de ser prolixo. Escrevi [um post gigante sobre a indústria em geral nesse link](http://lucasfcosta.com/2017/07/17/The-Ultimate-Guide-to-JavaScript-Fatigue.html) e tenho [uma talk sobre isso nesse outro link](https://www.youtube.com/watch?v=szliuFBcgX0).

**Comment by renanlara** [comment]
Nas minhas experiências, só fiquei 3 anos em uma empresa porque era sócio, mas valeu muuuito!
Depois, fiquei 1 ano e meio, sair pois fui selecionado em outra empresa que trabalhei por 1 ano. Após isso, fiquei um pouco mais de 1 ano atendendo diversas agências remotamente. Agora faz 1 ano que estou em um agência e já estou de saída hehe. 

Pra mim é o seguinte: se eu não estou gostando por motivo X, eu devo sair. Trabalhar em um lugar onde eu não me sinto bem, não me sinta confortável e não esteja valendo a pena é melhor procurar outra vaga.

**Comment by ninetails** [comment]
Pessoalmente eu também sou do time que não gosta de ficar "pulando de emprego". Mas reforçando: opinião pessoal, não daria motivos com base fundamentada, mas um motivo que mais me reforça pensando como empresa é de que se a pessoa não é estável em algum lugar, o que me passa é que a mesma, se eu a contratar, irá sair pela primeira oportunidade que pague melhor ou tenha algum diferencial que minha empresa não teria. Claro, excluindo contratos com tempo pré definidos.

Mas não acho que o mercado pense assim não, mesmo pq além do nosso mercado estar aquecido e com falta de bons profissionais, certamente também é de alta rotatividade.

É que depende do foco do profissional, uns preferem ganhar mais, outros aprender, tem workaholics como tem aqueles que batem cartão pontualmente. Acho muito mais interessante tomar o tempo como uma medida entre o quanto o profissional se adequa a cultura/filosofia da empresa e definir o quanto a pessoa quer ficar para atingir seus objetivos pessoais e profissionais enquanto trabalha nela.

**Comment by barbier** [comment]
Olha, por experiência eu endosso o que o @lucasfcosta falou.
Trabalhei por 6 meses em dois lugares onde sempre que eu volto a pensar sobre eles o pensamento é sempre o mesmo "Eu não deveria ter ido pra lá" e isso por motivos diversos que não cabem falar aqui, mas quem me conhece já deve ter ouvido as histórias.
Eu procuro ficar pelo menos um ano em cada lugar, acho uma boa janela para fazer uma boa avaliação sobre o quanto eu estou evoluindo com os projetos da empresa, tanto financeiramente quanto intelectualmente, e o quanto eu estou conseguindo dar de retorno para ela.

**Comment by lucasfcosta** [comment]
@ninetails gostaria de refutar alguns dos seus pontos pra levantar alguns questionamentos:

> mas um motivo que mais me reforça pensando como empresa é de que se a pessoa não é estável em algum lugar, o que me passa é que a mesma, se eu a contratar, irá sair pela primeira oportunidade que pague melhor ou tenha algum diferencial que minha empresa não teria

E isso é errado? O que há de errado em querer um salário melhor ou diferenciais que sua empresa não pode oferecer?

Under late capitalism, o que importa é oferta e demanda. Assim como sua empresa vai buscar os melhores profissionais, os profissionais vão buscar os melhores empregos. Se você como empresa quiser manter bons profissionais você precisa aumentar salários ou oferecer outros benefícios. Relações trabalhistas são muito diferentes de relações de amizade.

É claro que muitas pessoas podem continuar em uma empresa porque estão emocionalmente investidas, gostam do ambiente ou simplesmente se sentem felizes lá por qualquer motivo, mas isso também deve ser considerado um diferencial.

> É que depende do foco do profissional, uns preferem ganhar mais, outros aprender

E não há nada de errado em nenhum dos dois. Bons profissionais amam aprender e fazem isso pelo simples prazer de adquirir conhecimento e é claro que isso reflete no seu salário, afinal você vale o valor que você produz (se souber dar um preço correto pro seu próprio trabalho e for um bom negociador, é claro).

> Tem workaholics como tem aqueles que batem cartão pontualmente

Nesse caso também é importante apontar que nenhum deles é melhor que o outro.

> Acho muito mais interessante tomar o tempo como uma medida entre o quanto o profissional se adequa a cultura/filosofia da empresa e definir o quanto a pessoa quer ficar para atingir seus objetivos pessoais e profissionais enquanto trabalha nela

O tempo pode até ser uma medida, mas se você não souber qual dos lados da relação trabalhista teve comportamentos adequados, você realmente vai saber julgar se o problema é o profissional ou a empresa? Se um profissional não se adequa à uma determinada cultura é culpa dele? Ele deveria ficar só para não "manchar" seu currículo com pouco tempo em uma empresa?

Como eu falei e reforço, a única medida que importa é quanto valor uma pessoa gera versus o preço que ela custa.

PS.: Sei que esses questionamentos podem parecer "duros" de certa forma, mas não acho que seja possível comentar realidades de mercado de uma maneira mais sensível que essa. Entendo seu ponto de vista e respeito, até porque pode ser válido em determinados casos e em certas áreas, afinal não existe ciência exata em RH/people management, mas não acredito que seja esse o modelo ideal para a indústria de software.

**Comment by kvnol** [comment]
> Under late capitalism, o que importa é oferta e demanda. Assim como sua empresa vai buscar os melhores profissionais, os profissionais vão buscar os melhores empregos.

Concordo totalmente com o @lucasfcosta e digo mais: da mesma que as empresas tratam os funcionários como uma caneta que é jogada fora quando acaba a tinta, nós como canetas também podemos buscar novas oportunidades (papéis) onde vamos ter espaço de escrever mais (ganhar mais dinheiro, conhecimento e satisfação).

Ou seja, quando eu não consigo mais tirar nada da empresa em que estou (em questão de conhecimento e crescimento), eu saio fora sem pensar duas vezes, pois a empresa faria o mesmo comigo.

Cheguei a [falar disso, ano passado, no Twitter](https://twitter.com/kvnol/status/851238533071335424):

![image](https://user-images.githubusercontent.com/3299130/41205692-5e27e8ee-6cce-11e8-91a2-daf535550239.png)

**Comment by lucasfcosta** [comment]
@kvnol não foi exatamente isso que eu quis dizer. Acho que o termo "sugar" é um pouco forte.

Eu diria que a relação deve apenas ser vista como uma via de duas mãos. Se a empresa te deixa, por exemplo, aprender e estudar no horário de trabalho eu acho justo que você ofereça algo em troca, assim como quando você precisa pegar uma folga durante a semana pra cuidar do seu filho e a empresa não reclamar, também acredito que seria ético da sua parte trabalhar em algum horário que não seja o habitual para uma entrega importante ou algo assim.

Anyway, tomem tudo com uma pitada de sal. Existem casos e casos.

**Comment by caioincau** [comment]
Depende muito, desde qu você não se sinta estagnado.

Eu sinceramente, como CLT, prefiro ficar pelo menos um ano, para poder ter férias.

**Comment by ninetails** [comment]
@lucasfcosta já que vc quis botar sal, até citou capitalismo, vou botar a pimentinha: para as empresas, profissionais são nada mais que commodities.

> Funcionário merece mais, paga-se mais.
> Boi gordo que dá mais carne também.

Se é pra ser duro, vamos ser: Escrever código, qualquer um pode fazer. (não encarem como algo pejorativo e sim que o mercado realmente necessita de diferenciais)

Não gostei da sua postura dualista de achar que o que eu falo é certo ou errado, apenas expus minha visão, minha contribuição e deixei bem claro que não julguei nada por certo ou errado. Só comentei que eu não gosto de pular de emprego, pois não acho que isso é saudável para desenvolver um software, Não acho que vou conseguir adquirir conhecimento o suficiente do core, demanda e/ou funcionalidades da empresa, tão pouco passaria por desafios onde poderia me provar e melhorar minhas capacidades no caso de ficar pouco tempo nela. Longe de dizer pras pessoas se acomodarem, por favor. Não defendo por tal ponto.

E que, na pele de contratador, não abriria o meu core para alguém que em 3 meses irá pro concorrente, nem que eu tenha que pagar por um "estrelinha", afinal preciso maximizar os lucros.

Da mesma forma que, na pele de contratado, sim, as pessoas devem sim tentar maximizar seus ganhos também.

**Comment by evansantos** [comment]
O que sempre me motivou a mudar de emprego foi me sentir estagnado e não poder ajudar as coisas evoluírem _(como tecnologia, como participar da comunidade)_, além de um ponto que é o quão me sinto em casa e o ambiente seja propicio _vulgo não tóxico onde um tende a tentar ser melhor que o outro_

**Comment by lucasfcosta** [comment]
@ninetails Não estou dizendo que é certo ou errado, entendo perfeitamente sua opinião, apenas levantei alguns questionamentos.

Inclusive concordo com:

> E que, na pele de contratador, não abriria o meu core para alguém que em 3 meses irá pro concorrente, nem que eu tenha que pagar por um "estrelinha", afinal preciso maximizar os lucros.

Porém o que eu estou dizendo é que o fator pra você tentar determinar se isso vai acontecer não pode ser o tempo anterior em outras empresas, pois existem muitos fatores que podem fazer alguém sair de um determinado emprego, não necessariamente pagamento.

De forma alguma acredito que sua opinião esteja "errada" ou que a minha esteja "certa" ([inclusive adicionei aqui um "alerta de minha opinião™️"](https://github.com/frontendbr/forum/issues/1114#issuecomment-395673486). Peço desculpas se me expressei uma forma que pudesse deixar margem pra essa interpretação ❤️

**Comment by lucasmahle** [comment]
Eu acho engraçado vocês falando de mudar de empresa na maior naturalidade.
Acho lindo ter várias ofertas de emprego, aqui em Chapecó o negócio é mais complicado, pois os empresários (alguns) fazem complô pra não tirar funcionário do outro, e sobra só empresas "menores" com vaga. Mas mesmo assim, a oferta é baixa.

Então no caso de Chapecó, quanto menos trocar de empresa melhor, diversos empresários mantem aquela ideia de vestir a camisa numa empresa até o fim da vida. 

O que me prende aonde eu trabalho é o salário, embora outras empresas apresentam um conforto maior, poucas conseguem competir no aspecto de remuneração.

E por fim, particularmente nunca troquei uma empresa, mudei por fatores internos, no primeiro caso pela empresa ter vendido o setor que eu trabalhava (e fui de brinde kk) e no segundo caso mudei de cidade.

**Comment by ninetails** [comment]
@lucasfcosta ah sim, realmente passei como se fosse exclusivamente pagamento, me expressei mal e nisso vc está certo. Era para ter dito sobre estatística e probabilidade, mas nvm... nois :D

É, no interior é diferente o cenário... diria até talvez por isso que estão surgindo mais pessoas pedindo por vagas remoto, que não deixa de ser diferencial em contratação.

**Comment by JonatasRicardo** [comment]
Pessoal, muito bom ler as respostas de vocês.
Uma coisa não se pode negar: na nossa área existe uma rotatividade maior que a média geral.

@lucasfcosta gostei muito do seu [post](http://lucasfcosta.com/2017/07/17/The-Ultimate-Guide-to-JavaScript-Fatigue.html).

Pelo que eu pude entender com vocês é que o importante é não perder o **timing** específico de cada oportunidade.
Para mim os fatores qualidade de vida e família são muito importantes. Dar valor ao que realmente importa é primordial.
Então acho que esse  **timing** pode ser identificado com a seguinte pergunta: **Estou jogando meu precioso tempo fora aqui nessa empresa?**
